kiali/kiali
# Format links for Grafana Tempo

Fixes #6531 

How to test it: 
- run `istio/tempo/install-tempo-env.sh` to install the environment
- Setup Kiali config: 

```
  tracing:
    enabled: true
    provider: "tempo"
    in_cluster_url: http://localhost:3200
    url: http://localhost:3200

...
  grafana:
    in_cluster_url: http://localhost:13000
    url: http://localhost:13000

```
(Grafana url needs to be set, as Tempo will use Grafana UI with the datasource)

- Create a Grafana Tempo Datasource. In Grafana, go to Connections > Select Tempo > Create Tempo datasources.
- Set up `default` and put a name, and a url. It should be the internal url for the tempo service `query-frontend` and port `3200`. 
![image](https://github.com/kiali/kiali/assets/49480155/7c1157fd-1055-49d3-baae-2739e722cb7a)

Go to an application in Kiali, click in Traces Tab, click in `View in tracing`. The result should be something similar to: 

![image](https://github.com/kiali/kiali/assets/49480155/da71922d-9d89-46ae-8b4a-ffbb00c00a1f)

Click on a Trace, in the `Trace Details`, click in "View in Tracing". The result should be something similar to: 

![image](https://github.com/kiali/kiali/assets/49480155/3e2ad478-ce97-4841-b42d-df233caa0a90)

Click in the `Span Details` tab, click in More Span Details, the result should be something similar to: 

![image](https://github.com/kiali/kiali/assets/49480155/0a5e5b1b-fe2b-41ce-aff2-bb5786954f68)

(It redirects to the trace, as I don't see the spans are bookmarkable). 

Note: There is the assumption that if the provider is Tempo, grafana will be used as the Tempo UI
